### PR TITLE
Fix breaking changes caused by version upgrade and rollback version of `androidx.biometric`

### PR DESCRIPTION
**Pull request by rafaeltonholo**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [ ] The PR is submitted to the `main` branch.
- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [ ] 🎬 I've attached a **screen recoding** of the changes. 

## What's changed?
- Fixes the breaking changes caused by version upgrade of:
    - AGP
    - Kover
    - Compose compiler
- Rollback the version of `androidx.biometric:biometric` from `1.4.0-alpha01` to `1.2.0-alpha05`. 
    - **Reason**: `1.4.0-alpha01` requires `compileSdk` version 35 and the app is using 34.
    - **Suggestion**: As the `compileSkd` upgrade usually requires more changes and closer attention, I would address that in a separate PR.

## Risk Factors

**What may go wrong if we merge your PR?**

The present changes shouldn't affect the behaviour of the app.

## Does this PR closes any GitHub Issues?

This present PR doesn't close any issue, but helps to address the following:

- Fix issues of #3240 

## Troubleshooting CI failures ❌

GitHub Actions failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

**Review by ILIYANGERMANOV — CHANGES_REQUESTED** [review]
Can you revert most of the changes and just edit `libs.versions.toml`? The other changes are breaking our automations and CI. Just picking the right library versions - upgrading the Android Gradle Plugin should be sufficient

**Comment by ILIYANGERMANOV** [comment]
@rafaeltonholo got it. Let's:
- Downgrade Paparazzi
- Fix Detekt errors manually (that shouldn't be that hard, I guess) If the error is in the legacy code, you can directly `@Suppress()`

The idea was:
- do the minimal changes to make the dependabot PR green and work
- the fixes must not break the CI or existing automations

**Comment by ILIYANGERMANOV** [comment]
Most version upgrade fixes require:
- downgrade of troublesome libraries, you did the hard job by finding the culprits 👏 Kudos!
- fixing Detekt errors for new checks added with the newer versions
- fixing version incompatibilities

**Review by ILIYANGERMANOV — APPROVED** [review]
OMG! My bad, sorry! Actually your changes introduce the new bundled Compose in K2 👏

**Comment by ILIYANGERMANOV** [comment]
@rafaeltonholo great job! Any ideas why the Integration tests are failing? They run some of the code on a real Android device

File: gradlew.bat [hunk 1]
```diff
@@ -43,11 +43,11 @@ set JAVA_EXE=java.exe
 %JAVA_EXE% -version >NUL 2>&1
 if %ERRORLEVEL% equ 0 goto execute
 
-echo.
-echo ERROR: JAVA_HOME is not set and no 'java' command could be found in your PATH.
-echo.
-echo Please set the JAVA_HOME variable in your environment to match the
-echo location of your Java installation.
+echo. 1>&2
```
Comment (ILIYANGERMANOV) [inline gradlew.bat:46]: Any idea why do we need this change.

**Comment by ILIYANGERMANOV** [comment]
At some point, we can get rid of Paparazzi in favor of https://developer.android.com/studio/preview/compose-screenshot-testing

File: gradlew.bat (hunk 1 above)
Comment (rafaeltonholo) [reply gradlew.bat:46]: That is automatically updated when we run the `./gradlew wrapper` to upgrade the wrapper version. We need to upgrade the wrapper to upgrade AGP

**Comment by rafaeltonholo** [comment]
> @rafaeltonholo great job! Any ideas why the Integration tests are failing? They run some of the code on a real Android device

I was looking into the lint issues before checking that. I still need to investigate why it is failing but looking at the error log, it might be related to the coroutine upgrade. I still need to check.

File: gradlew.bat (hunk 1 above)
Comment (ILIYANGERMANOV) [reply gradlew.bat:46]: Got it. Thanks for the info!

**Comment by ILIYANGERMANOV** [comment]
Lint is easy to fix - some strings issues:

Screenshot_20240619_002828_Chrome
Screenshot_20240619_002840_Chrome

**Comment by rafaeltonholo** [comment]
> Lint is easy to fix - some strings issues:
> 
> Screenshot_20240619_002828_Chrome Screenshot_20240619_002840_Chrome

I was about to comment about that... It isn't too easy to fix, at least not for me as I don't speak Chinese haha
With the upgrade of the AGP to 8.5.0, the lint starts to fail because Android Lint detected that `note_you_are_trying_to_change_the_account_associated_with_the_loan` doesn't have any string parameter in the `zh-rCN` resource. 

I have no idea where to put the parameter in this case and I didn't want to use a translator since it could provide a false positive fix.

Maybe someone in the telegram could provide the correct translated string?

**Comment by ILIYANGERMANOV** [comment]
@rafaeltonholo got it 😄 If you have ChatGPT or Gemini, you can ask it to put the String parameter at the correct place

**Comment by ILIYANGERMANOV** [comment]
Worst case: just hack ` %s` string parameter at the end

**Comment by ILIYANGERMANOV** [comment]
@rafaeltonholo I found the string for you using ChatGPT:
```
<string name="note_you_are_trying_to_change_the_account_associated_with_the_loan">
    注意：您正在尝试将贷款 %s 关联的账户更改为不同货币的账户\n所有贷款记录将根据当天的汇率重新计算
</string>
```

Please fix new lines

**Comment by rafaeltonholo** [comment]
I finally found out why the integration test was failing. There is a change in the `File.writeText(String, Charset)` from Kotlin `1.9.23` to `2.0.0`.

In Kotlin 1.9.23:
```kotlin
public fun File.writeText(text: String, charset: Charset = Charsets.UTF_8): Unit = writeBytes(text.toByteArray(charset))
```
In Kotlin 2.0.0:
```kotlin
public fun File.writeText(text: String, charset: Charset = Charsets.UTF_8): Unit =
    FileOutputStream(this).use { it.writeTextImpl(text, charset) }
```

Because of that, the integration test started to fail when trying to create the json data fail. 

To fix the issue, that would be the required change:
```diff
    private fun createJsonDataFile(jsonString: String): File {
        val fileNamePrefix = "data"
        val fileNameSuffix = ".json"
        val outputDir = context.cacheDir

        val file = File.createTempFile(fileNamePrefix, fileNameSuffix, outputDir)
--        file.writeText(jsonString, Charsets.UTF_16)
++        FileOutputStream(file).use { it.write(jsonString.toByteArray(Charsets.UTF_16)) }

        return file
}
```

This is basically a replacement of the old implementation of `public fun File.writeText(text: String, charset: Charset = Charsets.UTF_8): Unit`

I'll test a bit more tomorrow before committing that change, but I would appreciate your inputs about that.

**Comment by ILIYANGERMANOV** [comment]
@rafaeltonholo great work! Can you fix the merge conflict and we can merge it?

**Pull request merged**